### PR TITLE
EUI-8601: JudicialUser field not working when used in a collection

### PR DESCRIPTION
**Pull request by Daniel-Lam**
### JIRA link (if applicable) ###
EUI-8601

### Change description ###
Re-implement `WriteJudicialUserField` component to extend `WriteComplexFieldComponent`, to ensure correct working when used in a collection.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by DavidJayakumar — APPROVED** [review]
LGTM

File: projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/judicial-user/write-judicial-user-field.component.html [hunk 1]
```diff
@@ -1,43 +1,21 @@
...
-      (onSelectionChange)="onSelectionChange(judicialUser)"
-    >
-      {{ judicialUser.fullName }} ({{ judicialUser.emailId }})
+  <input class="form-control bottom-30" [id]="id()"
+    [formControl]="judicialUserControl" [matAutocomplete]="autoComplete">
+  <mat-autocomplete autoActiveFirstOption #autoComplete="matAutocomplete" [displayWith]="displayJudicialUser">
+    <mat-option *ngFor="let judicialUser of filteredJudicialUsers$ | async"
+      class="select-option" [ngClass]="{'hide-autocomplete': !showAutocomplete}"
+      [value]="judicialUser" (onSelectionChange)="onSelectionChange($event)">
+      {{ judicialUser.fullName }}
```
Comment (johnbenjamin-hmcts) [inline projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/judicial-user/write-judicial-user-field.component.html:16]: @Daniel-Lam - we have to display in the format _judicialUser.fullName (judicialUser.Email)_

**Review by johnbenjamin-hmcts — CHANGES_REQUESTED** [review]
@Daniel-Lam - one minor change required.

File: projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/judicial-user/write-judicial-user-field.component.html (hunk 1 above)
Comment (Daniel-Lam) [reply projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/judicial-user/write-judicial-user-field.component.html:16]: Hi @johnbenjamin-hmcts - ah, thanks for that... I'll correct that now and re-release

**Review by johnbenjamin-hmcts — APPROVED** [review]

**Review by DavidJayakumar — DISMISSED** [review]
@Daniel-Lam Approved

**Comment by Daniel-Lam** [comment]
Superseded by PR #1532.

**Pull request closed without merging**